llvm/lib/Target/X86/X86RegisterInfo.cpp

```diff
@@ -172,7 +172,8 @@ X86RegisterInfo::getLargestLegalSuperClass(const TargetRegisterClass *RC,
       if (getRegSizeInBits(*Super) == getRegSizeInBits(*RC))
         return Super;
     }
-    Super = *I++;
+    Super = I.isValid() ? getRegClass(*I) : nullptr;
+    ++I;
```

Comment:
Are you allowed to increment an invalid iterator?

Reply:
In practice it won't break anything here, since the iterator is just a pointer. This whole code path is somewhat awkward, but I don't want to mess with it too much without understanding the underlying logic.